src/main.rs

```diff
@@ -148,28 +151,36 @@ pub(crate) extern "C" fn main(hart_id: usize, device_tree_blob_addr: usize) -> !
 }
 
 fn main_loop(ctx: &mut VirtContext, mctx: &mut MiralisContext) -> ! {
-    let mut bench = Benchmark::new();
-
     loop {
         unsafe {
             if config::BENCHMARK {
-                bench.start();
+                let mut bench = BENCH.lock();
+                bench.start_interval_counter(IntervalCounter::ExecutionTime);
+                bench.start_interval_counter(IntervalCounter::InstructionRet);
             }
 
             Arch::run_vcpu(ctx);
 
             if config::BENCHMARK {
-                bench.stop("main::run_vcpu");
-                bench.start();
+                let mut bench = BENCH.lock();
+                bench.stop_interval_counter(IntervalCounter::ExecutionTime, "main::run_vcpu");
+                bench.stop_interval_counter(IntervalCounter::InstructionRet, "main::run_vcpu");
             }
 
             handle_trap(ctx, mctx);
 
             if config::BENCHMARK {
-                if config::BENCHMARK_NB_EXITS {
-                    Benchmark::record_entry("nb_exits", "nb_exits", ctx.nb_exits);
+                let mut bench = BENCH.lock();
+                bench.stop_interval_counter(IntervalCounter::ExecutionTime, "main::handle_trap");
+                bench.stop_interval_counter(IntervalCounter::InstructionRet, "main::handle_trap");
+                bench.increment_counter(Counter::TotalExits);
+
+                // TODO : remove this and call from an ecall
+                if ctx.nb_exits % 50 == 0 {
```

Comment:
The number of exits before printing should probably be something that can be configured, just in case that value does not fit all use-cases.

Reply:
I removed the block as counters will probably logged at the end of the execution by ecall to Miralis.